workspaces/optic-engine/src/shapes/traverser.rs

```diff
@@ -287,6 +289,38 @@ impl ShapeTrail {
     new_trail.path.push(component);
     new_trail
   }
+
+  pub fn with_optional(&self, optional_shape_id: ShapeId, subject_shape_id: ShapeId) -> Self {
```

Comment:
Is this used / should we use this (along side with_nullable) / or should we remove this?

Reply:
Nice catch! Removing, part of a previous iteration of determining which shape to edit by mutating the trail. Didn't work out well 😅